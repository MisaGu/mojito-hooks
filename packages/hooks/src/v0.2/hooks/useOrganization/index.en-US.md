---
nav:
  path: /hooks
  group: Mojito
---

# useOrganization

Returns the current user `id` and the organization the user belongs to that match the specified filter (similarly
to `useProfile(...)`).


## Examples

<code src="./demo/demo1.tsx" />
