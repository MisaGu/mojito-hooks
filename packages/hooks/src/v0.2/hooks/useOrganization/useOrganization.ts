import { EMojitoQueries } from '../../domain/gql/queries';
import { IMojitoProfileResponse } from '../../domain/interfaces';
import { BaseQueryHookPropsWithForce } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

function transformFn(response?: IMojitoProfileResponse) {
  if (!response) return undefined;

  return response.me.userOrgs[0];
}

export type UseOrganizationData = ReturnType<typeof transformFn>;

export type UseOrganizationReturn = ReturnType<typeof useOrganization>;

export type UseOrganizationProps = BaseQueryHookPropsWithForce<UseOrganizationData>;

export function useOrganization({ force, options }: UseOrganizationProps = {}) {
  return useMojitoFactory({
    as: 'organization',
    query: EMojitoQueries.organization,
    options,
    transformFn,
    force,
    onlyAuthenticated: true,
  });
}

// legacy function
/*

export function useOrganization(options: UseQueryOptions<any> = {}): {
  organization: (IMojitoProfileUserOrg & IMojitoProfileCustomOrgs) | null;
} & ReturnType<typeof useMojito> {
  const result = useMojito({
    query: EMojitoQueries.organization,
    options,
    onlyAuthenticated: true,
  });

  const profile = result?.data?.me;
  const organization = profile?.userOrgs?.[0];

  useEffect(() => {
    if (!profile || !organization) return;

    Sentry.setUser({ id: profile.id });
    Sentry.setTag('profileID', profile.id);
    Sentry.setTag('organizationID', organization.id);
  }, [profile, organization]);

  return {
    organization,
    ...result,
  };
}


*/
