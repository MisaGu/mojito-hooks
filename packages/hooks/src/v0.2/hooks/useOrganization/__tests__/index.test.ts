import { useOrganization } from '../useOrganization';

describe('useToggle', () => {
  it('should be defined', () => {
    expect(useOrganization).toBeDefined();
  });
});
