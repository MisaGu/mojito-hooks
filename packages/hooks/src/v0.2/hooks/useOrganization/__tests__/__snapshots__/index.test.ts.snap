// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOrganization() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "organization": Object {
    "avatar": null,
    "bidAllowed": false,
    "externalUserId": "   ",
    "hasNotifications": false,
    "id": "mock-organization-to-user-id",
    "kycStatus": "",
    "notifications": Object {
      "completeYourProfile": false,
      "contactUs": false,
      "isTransactionalWithID": false,
      "uploadID": false,
    },
    "organizationId": "mock-organization-id",
    "role": "Admin",
    "settings": Object {
      "hasCompletedOnboarding": false,
      "notifications": Object {
        "bidOnSold": false,
        "savedBidOn": false,
        "savedSold": false,
      },
      "privacy": Object {
        "hideActivity": false,
        "showCollection": false,
        "showSaved": false,
      },
    },
    "username": null,
  },
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": true,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
